isis/src/mro/apps/crism2isis/crism2isis.cpp

```diff
@@ -0,0 +1,173 @@
+#include "crism2isis.h"
+#include "ProcessImportPds.h"
+#include "UserInterface.h"
+#include "Pvl.h"
+#include "FileName.h"
+#include "TextFile.h"
+
+using namespace std;
+
+namespace Isis{
+  void crism2isis(UserInterface &ui, Pvl *log) {
+    ProcessImportPds p;
+    Pvl pdsLabel;
+    PvlGroup results;
+
+    FileName inFile = ui.GetFileName("FROM");
+
+    p.SetPdsFile(inFile.expanded(), "", pdsLabel);
+    // 65535 is set to NULL
+    p.SetNull(65535, 65535);
+
+    CubeAttributeOutput &att = ui.GetOutputAttribute("TO");
+    Cube *ocube = p.SetOutputCube(ui.GetFileName("TO"), att);
+
+    Pvl outLabel;
+
+    Pvl labelPvl(inFile.expanded());
+
+    QString prodType;
+
+    if (labelPvl.hasKeyword("PRODUCT_TYPE")) {
+      prodType = (QString)labelPvl.findKeyword("PRODUCT_TYPE");
+    }
+    else {
+      QString msg = "Unsupported CRISM file type, supported types are: DDR, MRDR, and TRDR";
+      throw IException(IException::User, msg, _FILEINFO_);
+    }
```

Comment:
Looks like there are a few of these failure cases that the original tests didn't hit. May be worth while to test them now. They appear at the ends of most of the if/else blocks